frontends/election-manager/src/components/print_all_ballots_button.tsx

```diff
@@ -0,0 +1,362 @@
+      result: 'User directed to reconnect printer.',
+      error: 'Printer disconnected.',
+    });
+  }, [logger, userRole]);
+
+  const onRendered = useCallback(async () => {
+    assert(typeof ballotIndex === 'number');
+    if (printLock.lock()) {
+      if (needsPrinter) {
+        printLock.unlock();
+        return onPrintError();
+      }
+      await printer.print({
+        sides: 'two-sided-long-edge',
+        copies: ballotCopies,
+      });
+      await logBallotsPrinted();
+      await advancePrinting();
+      printLock.unlock();
+    }
+  }, [
+    ballotIndex,
+    printLock,
+    needsPrinter,
+    printer,
+    ballotCopies,
+    logBallotsPrinted,
+    advancePrinting,
+    onPrintError,
+  ]);
+
+  let mainContent = null;
+  let actions = null;
+  let onOverlayClick: VoidFunction | undefined = closeModal;
+
+  switch (modalState) {
+    case 'no-printer':
+      mainContent = (
+        <Prose>
+          <h1>
+            {printFailed ? 'Printer Disconnected' : 'No Printer Detected'}
+          </h1>
+          <p>
+            {printFailed
+              ? 'Printing stopped because the printer was disconnected. Reconnect the printer to try again.'
+              : 'Please connect the printer to print all ballot styles.'}
+          </p>
+        </Prose>
+      );
+      actions = <LinkButton onPress={closeModal}>Cancel</LinkButton>;
+      break;
+
+    case 'options':
+      mainContent = (
+        <Prose>
+          <h1>Print All Ballot Styles</h1>
+          <p>Select the ballot type and number of copies to print:</p>
+          <CenteredOptions>
+            {!draftMode && (
+              <p>
+                <BallotModeToggle
+                  ballotMode={ballotMode}
+                  setBallotMode={setBallotMode}
+                />
+              </p>
+            )}
+            <p>
+              <BallotTypeToggle
+                isAbsentee={isAbsentee}
+                setIsAbsentee={setIsAbsentee}
+              />
+            </p>
+            <p>
+              Copies{' '}
+              <BallotCopiesInput
+                ballotCopies={ballotCopies}
+                setBallotCopies={setBallotCopies}
+              />
+            </p>
+          </CenteredOptions>
+        </Prose>
+      );
+      actions = (
+        <React.Fragment>
+          <Button
+            primary
+            onPress={() => startPrint()}
+            warning={!draftMode && ballotMode !== BallotMode.Official}
+          >
+            <PrintBallotButtonText
+              ballotCopies={ballotCopies * ballotStyles.length}
+              ballotMode={ballotMode}
+              isAbsentee={isAbsentee}
+              election={election}
+            />
+          </Button>
+          <LinkButton onPress={closeModal}>Cancel</LinkButton>
+        </React.Fragment>
+      );
+      break;
+
+    case 'printing': {
+      assert(typeof ballotIndex === 'number');
```

Comment:
The purpose of this assertion might be clearer if you instead use:

```ts
assert(ballotIndex !== undefined);
```

I could see someone wondering why this is necessary when the state value has a `number` type